runtime/integration-tests/src/generic/envs/runtime_env.rs

```diff
@@ -46,6 +55,258 @@ impl<T: Runtime> Default for RuntimeEnv<T> {
+			.get(&name.into())
+			.expect("Not deployed")
+			.clone()
+	}
+
+	fn deploy(
+		&mut self,
+		what: impl Into<String>,
+		name: impl Into<String>,
+		who: Keyring,
+		args: Option<&[Token]>,
+	) {
+		let info = self
+			.sol_contracts
+			.as_ref()
+			.expect("Need to load_contracts first")
+			.get(&what.into())
+			.expect("Unknown contract")
+			.clone();
+
+		let init = match (info.contract.constructor(), args) {
+			(None, None) => info.bytecode.to_vec(),
+			(Some(constructor), Some(args)) => constructor
+				.encode_input(info.bytecode.to_vec(), args)
+				.expect(ESSENTIAL),
+			(Some(constructor), None) => constructor
+				.encode_input(info.bytecode.to_vec(), &[])
+				.expect(ESSENTIAL),
+			(None, Some(_)) => panic!("Contract expects constructor arguments."),
+		};
+
+		let create_info = self.parachain_state_mut(|| {
+			let (base_fee, _) = <T as pallet_evm::Config>::FeeCalculator::min_gas_price();
+
+			<T as pallet_evm::Config>::Runner::create(
+				who.into(),
+				init,
+				0u8.into(),
+				GAS_LIMIT,
+				Some(base_fee),
+				None,
+				None,
+				Vec::new(),
+				// NOTE: Taken from pallet-evm implementation
+				VALIDATE,
+				// NOTE: Taken from pallet-evm implementation
+				TRANSACTIONAL,
+				None,
+				None,
+				<T as pallet_evm::Config>::config(),
+			)
+			.expect(ESSENTIAL)
+		});
+
+		let runtime_code =
+			self.parachain_state(|| pallet_evm::AccountCodes::<T>::get(create_info.value));
+
+		assert_eq!(runtime_code, info.deployed_bytecode);
+
+		self.deployed_contracts.insert(
+			name.into(),
+			DeployedContractInfo::new(info.contract.clone(), runtime_code, create_info),
+		);
+	}
+
+	fn call(
+		// TODO: Needs to imutable actually, but the current state implementation does
+		//       not rollback but error out upon changes, which is not ideal if you want to
+		//       test stuff without altering your state.
+		&mut self,
+		caller: Keyring,
+		value: U256,
+		contract: impl Into<String>,
+		function: impl Into<String>,
+		args: Option<&[Token]>,
+	) -> Result<CallInfo, DispatchError> {
+		self.call(caller, value, contract, function, args)
+	}
+
+	fn call_mut(
+		&mut self,
+		caller: Keyring,
+		value: U256,
+		contract: impl Into<String>,
+		function: impl Into<String>,
+		args: Option<&[Token]>,
+	) -> Result<CallInfo, DispatchError> {
+		self.call_mut(caller, value, contract, function, args)
+	}
+
+	fn view(
+		&mut self,
+		caller: Keyring,
+		contract: impl Into<String>,
+		function: impl Into<String>,
+		args: Option<&[Token]>,
+	) -> Result<CallInfo, DispatchError> {
+		self.call(caller, U256::zero(), contract, function, args)
+	}
+}
+
+impl<T: Runtime> RuntimeEnv<T> {
+	pub fn call_mut(
+		&mut self,
+		caller: Keyring,
+		value: U256,
+		contract: impl Into<String>,
+		function: impl Into<String>,
+		args: Option<&[Token]>,
+	) -> Result<CallInfo, DispatchError> {
+		let contract_info = self
+			.deployed_contracts
+			.get(&contract.into())
+			.expect("Contract not deployed")
```

Comment:
Nit - return a `DispatchError::Other`.

Reply:
I think the idea here is to return errors only from EVM side. Note that this is test code, and you want to panic as close to the origin of the error as possible if this method is called wrongly.